Sources/TuistCache/Cache/FileDownloader.swift

```diff
@@ -0,0 +1,85 @@
+import Foundation
+import RxSwift
+import TSCBasic
+
+enum FileDownloaderError: LocalizedError {
+    case noLocalURL
+    case invalidResponse
+    case urlSessionError(Error)
+    case moveFileError(Error)
+
+    var errorDescription: String? {
+        switch self {
+        case .noLocalURL:
+            return "Didn't receive a valid local URL"
+        case .invalidResponse:
+            return "Invalid URL response"
```

Comment:
I think we could include the HTTP status code and the response body to facilitate debugging.

Reply:
`FileDownloader.swift` isn't ready and hasn't been properly tested yet. It's my bad! I'll remove it from this PR and work more on it later ;)